### ISSUE TITLE
Tax does not work

### ISSUE THREAD

**Issue by EvilOlaf** [ChestShop-authors/ChestShop-3#68]
```
[10:30:51 ERROR]: Could not pass event CurrencyAddEvent to ChestShop v3.8.13-SNAPSHOT (build #26)
org.bukkit.event.EventException
>       at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at com.Acrobot.ChestShop.ChestShop.callEvent(ChestShop.java:404) [ChestShop-b26.jar:?]
>       at com.Acrobot.ChestShop.Listeners.Economy.TaxModule.onCurrencyAdd(TaxModule.java:50) [ChestShop-b26.jar:?]
>       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
>       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
>       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
>       at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
>       at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at com.Acrobot.ChestShop.ChestShop.callEvent(ChestShop.java:404) [ChestShop-b26.jar:?]
>       at com.Acrobot.ChestShop.Listeners.PostTransaction.EconomicModule.onBuyTransaction(EconomicModule.java:28) [ChestShop-b26.jar:?]
>       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
>       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
>       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
>       at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
>       at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at com.Acrobot.ChestShop.Listeners.Player.PlayerInteract.onInteract(PlayerInteract.java:107) [ChestShop-b26.jar:?]
>       at sun.reflect.GeneratedMethodAccessor214.invoke(Unknown Source) ~[?:?]
>       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
>       at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
>       at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:228) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.PlayerInteractManager.interact(PlayerInteractManager.java:463) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:759) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:52) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:1) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_131]
>       at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_131]
>       at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:715) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:374) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:654) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:557) [spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: java.lang.IllegalArgumentException: UUID cannot be null
>       at org.apache.commons.lang.Validate.notNull(Validate.java:192) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.craftbukkit.v1_8_R3.CraftServer.getOfflinePlayer(CraftServer.java:1318) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at org.bukkit.Bukkit.getOfflinePlayer(Bukkit.java:759) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       at com.Acrobot.ChestShop.Listeners.Economy.Plugins.VaultListener.onCurrencyAdd(VaultListener.java:139) ~[?:?]
>       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
>       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
>       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
>       at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
>       at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot.jar:git-Spigot-21fe707-e1ebe52]
>       ... 45 more

```

Occurs when `TAX_AMOUNT:` set to something else than `0` and the server account does not have an UUID (so new account created by the economy plugin).

build 26
Spigot 1.8.8 in offline mode, running behind Bungee in online mode.

**Comment by Phoenix616** [comment]
This happens if you set the server economy account to an unknown player. There is even a warning on startup/reload: `Server economy account setting 'admin' doesn't seem to be the name of a known player! Please log in at least once in order for the server economy account to work.`

What do you expect to happen? (besides the error message not getting thrown) That the tax doesn't work at all or that it will still remove the tax from the value but doesn't add it to the account. (same way like admin shops work with an invalid account)

**Comment by Phoenix616** [comment]
Fixed in commit 29ef7f2. (build 39)
